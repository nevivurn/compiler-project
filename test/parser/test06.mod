//
// test06
//
// parser test:
// - integer & boolean data types
// - binary and unary expressions
//
module test06;

var a,b,c: integer;
    b0,b1: boolean;

procedure unary(a: integer; b: boolean);
var x: integer;
    y: boolean;
begin
  x := -a;
  y := !b
end unary;

procedure binary(a: integer; b: boolean);
var x: integer;
    y: boolean;
begin
  x := -a * 3;
  y := !b || b
end binary;

begin
end test06.
