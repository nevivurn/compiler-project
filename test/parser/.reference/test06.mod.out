parsing 'test06.mod'...
successfully parsed.
  AST:
    CAstScope: 'test06'
      symbol table:
        [[
          [ *DIM(<ptr(8) to <NULL>>,<integer>) --> <integer>           ]
          [ *DOFS(<ptr(8) to <NULL>>) --> <integer>           ]
          [ *ReadInt() --> <integer>           ]
          [ *ReadLong() --> <longint>           ]
          [ *WriteChar(<char>) --> <NULL>           ]
          [ *WriteInt(<integer>) --> <NULL>           ]
          [ *WriteLn() --> <NULL>           ]
          [ *WriteLong(<longint>) --> <NULL>           ]
          [ *WriteStr(<ptr(8) to array of <char>>>) --> <NULL>           ]
          [ @a        <integer>           a           ]
          [ @b        <integer>           b           ]
          [ @b0       <boolean>          b0           ]
          [ @b1       <boolean>          b1           ]
          [ *binary(<integer>,<boolean>) --> <NULL>           ]
          [ @c        <integer>           c           ]
          [ main     <NULL>           ]
          [ *unary(<integer>,<boolean>) --> <NULL>           ]
        ]]
      statement list:
        empty.
      nested scopes:
        CAstScope: 'unary'
          symbol table:
            [[
              [ %a        <integer>               ]
              [ %b        <boolean>               ]
              [ $x        <integer>               ]
              [ $y        <boolean>               ]
            ]]
          statement list:
            := <integer>
              [ $x        <integer> ] <integer>
              neg <integer>
                [ %a        <integer> ] <integer>
            := <boolean>
              [ $y        <boolean> ] <boolean>
              not <boolean>
                [ %b        <boolean> ] <boolean>
          nested scopes:
            empty.
        
        CAstScope: 'binary'
          symbol table:
            [[
              [ %a        <integer>               ]
              [ %b        <boolean>               ]
              [ $x        <integer>               ]
              [ $y        <boolean>               ]
            ]]
          statement list:
            := <integer>
              [ $x        <integer> ] <integer>
              neg <integer>
                mul <integer>
                  [ %a        <integer> ] <integer>
                  3 <integer>
            := <boolean>
              [ $y        <boolean> ] <boolean>
              or <boolean>
                not <boolean>
                  [ %b        <boolean> ] <boolean>
                [ %b        <boolean> ] <boolean>
          nested scopes:
            empty.
        
    


run the following command to convert the .dot file into a PDF:
  dot -Tpdf -otest06.mod.ast.pdf test06.mod.ast.dot


Done.
